src/app/controllers/location/create.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name flightNodeApp.controller.location:LocationCreateController
+ * @description
+ * # LocationCreateController
+ * Controller for the create work type page.
+ */
+angular.module('flightNodeApp')
+	.controller('LocationCreateController',
+		['$scope', '$http', '$log', '$location', 'messenger', 'oauthRequest',
+			function ($scope, $http, $log, $location, messenger, oauthRequest) {
+
+				$scope.loading = true;
+
+				$scope.cancel = function () {
+					$location.path('/locations');
+				}
+
+				$scope.submit = function () {
+					$scope.loading = true;
+
+					oauthRequest.post('http://localhost:50323/api/v1/locations', $scope.location)
+						.then(function success(response) {
+
+							messenger.showSuccessMessage($scope, 'Saved');
+							$scope.loading = false;
+
+						}, function error(response) {
+
+							switch (response.status) {
+								case 400:
+									var messages = [{ error: response.data.message }];
+									if (response.data.modelState) {
+										_.forIn(response.data.modelState, function (value, key) {
+											messages.push({ error: value.toString() });
+										});
+									}
+									messenger.showErrorMessage($scope, messages);
```

Comment:
This does have the risk of exposing more internal details than i would like to the user, so it needs to be fine-tuned.